docs/how-to-guides/develop-a-tool/add-a-tool-icon.md

```diff
@@ -0,0 +1,64 @@
+# Add a Tool Icon
+A tool icon serves as a graphical representation of your tool in the user interface (UI). Follow this guidance to add a custom tool icon when developing your own tool package.  
+
+Adding a custom tool icon is optional. If you do not provide one, the system uses a default icon.
+
+## Prerequisites
+
+- Create a tool project as described in [Create and Use Tool Package](create-and-use-tool-package.md).
+
+- Install the dependent library:
+
+  ```
+  pip install pillow
+  ```
+
+- Prepare a custom tool icon image in `PNG` or `JPG` format.
+
+  The script in the next section verifies the image size and resizes it to 16x16 pixels if needed. To maintain quality:
+  - Provide a 16x16 pixel image to avoid distortion from resizing.
+  - Avoid complex images with much detail or contrast as they may not resize well.
+
+
+## Generate a tool icon data URI
+Generate a data URI for your custom tool icon: 
+```
+python <path-to-scripts>\tool\convert_image_to_data_url.py --image-path <image_input_path> -o <html_output_path>
+```    
+Ensure the output file has a `.html` extension.   
+For example:
+```
+python D:\proj\github\promptflow\scripts\tool\convert_image_to_data_url.py --image-path D:\proj\github\promptflow\examples\tools\tool-package-quickstart\my_tool_package\icons\custom-tool-icon.png -o output.html
+```
+The `output.html` content resembles:
```

Comment:
The content of `output.html` looks like the following, and you can open it in a web browser to see the result.

Reply:
Changed, thanks